packages/framework/src/Model/AdminNavigation/SideMenuBuilder.php

```diff
@@ -148,7 +148,9 @@ protected function createPricingMenu(): ItemInterface
         $menu->addChild('vat', ['route' => 'admin_vat_list', 'label' => t('VAT and rounding')]);
         $menu->addChild('free_transport_and_payment', ['route' => 'admin_transportandpayment_freetransportandpaymentlimit', 'label' => t('Free shipping and payment')]);
         $menu->addChild('currencies', ['route' => 'admin_currency_list', 'label' => t('Currencies')]);
-        $menu->addChild('promo_codes', ['route' => 'admin_promocode_list', 'label' => t('Promo codes')]);
+        $promoCodesMenu = $menu->addChild('promo_codes', ['route' => 'admin_promocode_pageeditlist', 'label' => t('Promo codes')]);
```

Comment:
I think that this might be considered BC break. You kept old page with promo codes, but changed link to new implementation. In this case it is highly possible, that people will be forced to new version.

If you would have use parameter you could even decide what version of menu would be displayed.

Reply:
I couldn't use it as I described above. The old page is deprecated so it shouldn't be used anymore.

Reply:
I tried to describe some concept in previous comment.